src/main/java/net/eldeen/dropwizard/CfSignalResourceBundle.java

```diff
@@ -7,75 +17,154 @@
-import io.dropwizard.Bundle;
-import io.dropwizard.lifecycle.ServerLifecycleListener;
+import com.google.common.base.Throwables;
+import io.dropwizard.Configuration;
+import io.dropwizard.ConfiguredBundle;
 import io.dropwizard.setup.Bootstrap;
 import io.dropwizard.setup.Environment;
-import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.util.component.LifeCycle;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Signal the AutoScalingGroup via CloudFormation Signal when running on an AWS EC2 instance.
  */
-public class CfSignalResourceBundle implements Bundle {
+public class CfSignalResourceBundle<T extends Configuration> implements ConfiguredBundle<T> {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(CfSignalResourceBundle.class);
 
-  private final String asgResourceName;
-  private final String stackName;
-  private final String awsRegion;
-
-  /**
-   * Create a bundle to signal the AutoScalingGroup via CloudFormation Signal
-   *
-   * @param asgResourceName the name of the auto scaling group cloudformation resource to signal
-   * @param stackName       the name of the stack containing the auto scaling group to signal
-   * @param awsRegion       the aws region the stack is in, e.g. us-west-2
-   */
-  public CfSignalResourceBundle(String asgResourceName, String stackName, String awsRegion) {
-    this.asgResourceName = asgResourceName;
-    this.stackName = stackName;
-    this.awsRegion = awsRegion;
+  private final Supplier<Optional<String>> instanceIdProvider;
+  private final Function<CfSignalResourceConfig, AmazonCloudFormation> cloudFormationSupplier;
+  private final AtomicReference<AmazonCloudFormation> cloudFormationAtomicRef = new AtomicReference<>(null);
+
+  public CfSignalResourceBundle() {
+    instanceIdProvider = () -> Optional.ofNullable(EC2MetadataUtils.getInstanceId());
+
+    cloudFormationSupplier = (cfSignalResourceConfig) -> {
+      AmazonCloudFormation amazonCloudFormation = cloudFormationAtomicRef.get();
+
+      if (amazonCloudFormation != null) {
+        return amazonCloudFormation;
+      }
+
+      return cloudFormationAtomicRef.updateAndGet((unused) -> {
+
+        AmazonCloudFormationClient amazonCloudFormationClient = new AmazonCloudFormationClient();
+
+        amazonCloudFormationClient.setRegion(Region.getRegion(Regions.valueOf(cfSignalResourceConfig.getAwsRegion()
+                                                                                                    .toUpperCase()
+                                                                                                    .replace('-', '_'))));
+        return amazonCloudFormationClient;
+      });
+    };
+  }
+
+  @Inject
+  public CfSignalResourceBundle(@CfSignalResourceInstanceId Optional<String> instanceId, AmazonCloudFormation amazonCloudFormation) {
+    instanceIdProvider = () -> checkNotNull(instanceId);
+    cloudFormationSupplier = (config) -> amazonCloudFormation;
```

Comment:
If you switch to Guava's Supplier / Function types, you can use `Supplier.ofInstance` and `Functions.constant` here, respectively.